src/component/component.ts

```diff
@@ -22,8 +22,8 @@ export type ComponentRenderProxy<P = {}, S = {}, PublicProps = P> = {
   S;
 
 // for Vetur and TSX support
-type VueConstructorProxy<PropsOptions, RawBindings> = {
-  new (): ComponentRenderProxy<
+type VueConstructorProxy<PropsOptions, RawBindings> = VueConstructor & {
+  new (...args: any[]): ComponentRenderProxy<
```

Comment:
Modified this type as it cannot match existing typings which expects `VueConstructor` like vue-router. By writing argument as `(...args: any[])`, we can mix it with the intersected constructor so that `ComponentRenderProxy` retains even after inherit `VueConstructor`.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-2.html

This probably fixes #185 ?